### PR TITLE
talk to inventory service

### PR DESCRIPTION
**Pull request by beav**
This commit does a few things:

 * reaches out to inventory service for facts based on the IDs sent in
 * "pivots" facts from inventory service into something usable by the
    web UI
 * returns a 400 error if `host_ids[]` is not present
 * returns a 400 error `x-rh-identity` header is not present

This commit also pins Travis tests at python 3.6.

### CODE REVIEW

File: drift/views/v0.py [hunk 1]
```diff
@@ -1,35 +1,30 @@
...
-    try:
-        r = requests.get(INVENTORY_SVC_URL + INVENTORY_SVC_PATH + '/health')
-        if r.status_code == requests.codes.ok:
-            inventory_connection_status = "pass"
-    except RequestException as re:
-        current_app.logger.warn("Unable to contact inventory service: %s" % re)
-        pass
-
-    return jsonify({'status': "running", 'inventory_connection': inventory_connection_status})
+    return jsonify({'status': "running"})
```
Comment (ehelms) [inline drift/views/v0.py]: Dropping the inventory status?

Comment (beav) [reply drift/views/v0.py]: ya, I had it there mostly as a baby step to get further on talking to the inventory service. I wasn't sure if I should leave it or not, but I figured we wouldn't want to get an alert on drift service if inventory service was down.

Comment (ehelms) [reply drift/views/v0.py]: Fair. We should think about if inventory service is unreachable how that affects the status of Drift given we rely heavily on inventory.

File: drift/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class BadRequest(Exception):
+    status_code = 400
```
Comment (ehelms) [inline drift/exceptions.py]: Trying to recall my python vs. ruby. Given this is a constant, should it be STATUS_CODE ?

File: drift/info_parser.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
+DRIFT_FACTS = {'bios_uuid', 'display_name'}
```
Comment (ehelms) [inline drift/info_parser.py]: Do we want to care about 'trimming' the list of facts vs. showing all available facts?

File: drift/info_parser.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    # TODO: we are assuming all info keys exist for each host. This is not a good assumption
+    available_info_names = ids_and_info[0].keys()
+
+    info_comparisons = [create_comparison(ids_and_info, info_name)
+                        for info_name in available_info_names
+                        if info_name in info_keys]
+    return info_comparisons
+
+
+def create_comparison(hosts_infos, info_name):
```
Comment (ehelms) [inline drift/info_parser.py]: The plural seems unneeded on `infos`. What is `info_name` in this respect?

File: drift/views/v0.py (hunk 1 above)
Comment (beav) [reply drift/views/v0.py]: I'll add a card for this. I think as long as we return a 500 and don't retry the connection from drift to inventory it should be OK.

File: drift/info_parser.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        info_comparison = 'SAME'
+
+    return {'name': info_name, 'status': info_comparison, 'hosts': host_values}
+
+
+def host_metadata(host):
+    return {'id': host['id'], 'fqdn': host['fqdn'], 'last_updated': host['updated']}
+
+
+def build_comparisons(hosts_from_inventory_svc):
```
Comment (ehelms) [inline drift/info_parser.py]: What is svc?

File: drift/inventory_service_interface.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    return incoming_headers.get(AUTH_HEADER_NAME)
+
+
+def fetch_hosts(host_ids, service_auth_key):
+    auth_header = {AUTH_HEADER_NAME: service_auth_key}
+    hosts = []
+
+    for host_id in host_ids:
+        response = requests.get(INVENTORY_SVC_URL + INVENTORY_SVC_PATH +
+                                '/hosts/' + host_id, headers=auth_header)
```
Comment (ehelms) [inline drift/inventory_service_interface.py]: The base URL with `hosts` path could be built outside of the list iteration and potentially stored as a constant above.

File: drift/views/v0.py [hunk 7]
```diff
@@ -1,35 +1,30 @@
...
 
 @section.route("/compare")
 def compare():
-    if 'X-RH-IDENTITY' in request.headers:
-        current_app.logger.debug("X-RH-IDENTITY received")
-    return jsonify(PHONY_DATA)
+    host_ids = request.args.getlist('host_ids[]')
+    auth_key = get_key_from_headers(request.headers)
+
+    if host_ids is None or host_ids == []:
```
Comment (ehelms) [inline drift/views/v0.py]: If I recall my python, this can be simplified to `if not host_ids`

File: tests/test_inventory_service_interface.py [hunk 8]
```diff
@@ -0,0 +1,59 @@
+from drift import inventory_service_interface
+import responses
+
+import unittest
+
+
+class InventoryServiceTests(unittest.TestCase):
+
+    HOST_TEMPLATE = '''
```
Comment (ehelms) [inline tests/test_inventory_service_interface.py]: Could this be moved out to a fixture file?

File: drift/info_parser.py (hunk 5 above)
Comment (beav) [reply drift/info_parser.py]: service. I guess i can flip it to be "inventory_service_hosts" or something

File: drift/views/v0.py (hunk 1 above)
Comment (beav) [reply drift/views/v0.py]: (card added, marking as resolved) :card_file_box:

File: drift/exceptions.py (hunk 2 above)
Comment (beav) [reply drift/exceptions.py]: This is some boilerplate from the flask docs on raising exceptions. However it still seems a little strange to me that BadRequest is something we have to define. I'll look into it more today.

File: drift/info_parser.py (hunk 3 above)
Comment (beav) [reply drift/info_parser.py]: When the data comes back, it's mostly various IDs that wouldn't make sense to compare. There's a `facts` list as well but it's empty currently.

I think eventually we'll just be using the `facts` list, but for now I use this set to create some ersatz facts based on the most easily recognized IDs :blowfish:

File: drift/info_parser.py (hunk 5 above)
Comment (ehelms) [reply drift/info_parser.py]: Ahh, I suck at mentally interpreting acronyms when I read code.

File: drift/info_parser.py (hunk 4 above)
Comment (beav) [reply drift/info_parser.py]: it is the key for the item coming back from inventory service. I originally called it `fact_name` but not all pieces of information coming back are facts.

I'll add a docblock to these methods, they are kinda tricky to understand

File: drift/views/v0.py (hunk 7 above)
Comment (beav) [reply drift/views/v0.py]: i just tried, you're right

File: drift/info_parser.py (hunk 4 above)
Comment (beav) [reply drift/info_parser.py]: docblocks added

File: drift/inventory_service_interface.py (hunk 6 above)
Comment (beav) [reply drift/inventory_service_interface.py]: updated

File: tests/test_inventory_service_interface.py (hunk 8 above)
Comment (beav) [reply tests/test_inventory_service_interface.py]: done

File: tests/test_v0_api.py [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+import mock
+import unittest
+
+
+class ApiTests(unittest.TestCase):
+
+    AUTH_HEADER = {'X-RH-IDENTITY': 'VGhlIEZhc3QgYW5kIHRoZSBGdXJpb3VzOiBUb2t5byBEcmlmdAo='}
+
+    FETCH_HOSTS_RESULT = [
```
[the remark this answers is not in the document]
Comment (beav) [reply tests/test_v0_api.py]: done

File: drift/info_parser.py (hunk 5 above)
Comment (beav) [reply drift/info_parser.py]: updtd 4 fwr acrnms

File: tests/test_inventory_service_interface.py (hunk 8 above)
Comment (ehelms) [inline tests/test_inventory_service_interface.py]: Assuming you meant to move this one as well?

File: drift/info_parser.py [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+
+def _host_metadata(host):
+    """
+    create a metadata field for one host
+    """
+    return {'id': host['id'], 'fqdn': host['fqdn'], 'last_updated': host['updated']}
+
+
+def build_comparisons(inventory_service_hosts):
```
Comment (ehelms) [inline drift/info_parser.py]: Nitpick -- move this to the top since its the `public` function to make it more obvious when reading the code?

**Comment by beav** [comment]
thanks @ehelms !

**Pull request merged**